jfrog-oauth/run.sh

```diff
@@ -47,12 +53,75 @@ index-url = https://${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}@${JFROG_
 EOF
 fi
 
-# Set GOPROXY to use the Artifactory "go" repository.
+# Configure Artifactory "go" repository.
 if [ -z "${REPOSITORY_GO}" ]; then
-  echo "🤔 REPOSITORY_GO is not set, skipping go configuration."
+  echo "🤔 no go repository is set, skipping go configuration."
+  echo "You can configure a go repository by providing the a key for 'go' in the 'package_managers' input."
 else
   echo "🐹 Configuring go..."
-  jf go-config --global --repo-resolve "${REPOSITORY_GO}"
-  export GOPROXY="https://${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}@${JFROG_HOST}/artifactory/api/go/${REPOSITORY_GO}"
+  jf goc --global --repo-resolve "${REPOSITORY_GO}"
 fi
 echo "🥳 Configuration complete!"
+
+# Configure the JFrog CLI to use the Artifactory "docker" repository.
+if [ -z "${REPOSITORY_DOCKER}" ]; then
+  echo "🤔 no docker repository is set, skipping docker configuration."
+  echo "You can configure a docker repository by providing the a key for 'docker' in the 'package_managers' input."
+else
+  echo "🐳 Configuring docker..."
+  mkdir -p ~/.docker
+  cat << EOF > ~/.docker/config.json
+{
+  "auths": {
+    "${JFROG_HOST}": {
+      "auth": "$(echo -n "${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}" | base64)",
+      "email": "${ARTIFACTORY_EMAIL}"
+    }
+  }
+}
+EOF
+fi
+
+# Install the JFrog vscode extension for code-server.
+if [ "${CONFIGURE_CODE_SERVER}" == "true" ]; then
+  while ! [ -x /tmp/code-server/bin/code-server ]; do
+    counter=0
+    if [ $counter -eq 30 ]; then
```

Comment:
yes, I agree. Having a mechanism for dependence is probably the best base. For now we can modify the code-server module to create status files. I am not sure how we can check the insatlling status.

Reply:
Also, should we condition the part of the module or the whole module? I think your idea can be extended to have a common input for all modules `depends_on = module_name` where we check for file `/tmp/.${module_name}-success` before we start runing `coder_script`

Reply:
If there's a way to reference another modules resource, I think that would work well for this use case. But I'm not sure how terraform would behave in this case. Maybe we need both a module and a script level dependency so that terraform can exit early if only one module is used without its dependency?

Or do we automatically pull in dependencies without them being expplicitly declared?

Anyway, as for how the script part might look, maybe something like this?

```tf
// code-server
resource "coder_script" "install" {}

// my-module
resource "coder_script" "takes_a_while" {}

resource "coder_script" "finalize" {
	depends_on = [code-server.coder_script.install, coder_script.takes_a_while]
}
```

Again, if we can refer like this, it'd be ideal, otherwise we may need to use strings and define the dependency in runtime? Perhaps it would be a good idea to think about what other use-cases we have, and maybe write a proposal? This might also tie a little bit into https://github.com/coder/coder/issues/11131?

Reply:
In the meantime, I'd rather not use status files, but instead wait for the process to be started or a successful curl or something.

Reply:
For now I am going to merge them without any dependence. Anyone?